django/forms/formsets.py

```diff
@@ -105,6 +105,33 @@ def management_form(self):
             })
         return form
 
+    def _extract_form_indices(self):
+        """Find indices of all submitted forms"""
+        prefix = self.prefix + '-'
+        prefix_len = len(prefix)
+        for key in self.data.keys():
+            if key.startswith(prefix):
+                identifier = key[prefix_len:].split('-', 1)[0]
+                if identifier.isdigit():
+                    yield int(identifier)
+
+    def _extract_total_form_count(self):
+        """Extract the number of submitted forms"""
+        if self.management_form.cleaned_data[TOTAL_FORM_COUNT]:
+            # prefer the management form if data is present
+            return self.management_form.cleaned_data[TOTAL_FORM_COUNT]
+        try:
+            return max(self._extract_form_indices()) + 1
+        except ValueError:  # no forms submitted
+            return 0
+
+    def _extract_initial_form_count(self):
+        """Extract the number of initial forms"""
+        if self.management_form.cleaned_data[INITIAL_FORM_COUNT]:
+            # prefer the management form if data is present
+            return self.management_form.cleaned_data[INITIAL_FORM_COUNT]
+        return 0
```

Comment:
Hmm, I'm not sure I see what is proven by wtforms, or how it can be said that "most day to day use cases" don't depend on the existence of the management form. All uses of formsets with any extra forms depend on the initial-form-count being set correctly (in order to handle `empty_permitted` correctly); that definitely seems like a "day to day use case" to me. In fact, in my experience inline-formsets are even more commonly used than plain formsets, and inline formsets depend even more heavily on initial-form-count.

So while I think "only maintain backward compatibility in the case where the management form is present" is fine, I don't really see how it's feasible to make the management form optional without providing an alternative _correct_ handling for initial-form-count and empty-permitted. (Unless you are suggesting that having a formset whose form has required fields is not a "day to day" use?)